core/src/test/scala/integration/kafka/api/RackAwareAutoTopicCreationTest.scala

```diff
@@ -37,29 +43,52 @@ class RackAwareAutoTopicCreationTest extends KafkaServerTestHarness with RackAwa
 
   def generateConfigs =
     (0 until numServers) map { node =>
-      TestUtils.createBrokerConfig(node, zkConnect, enableControlledShutdown = false, rack = Some((node / 2).toString))
+      TestUtils.createBrokerConfig(node, zkConnectOrNull, enableControlledShutdown = false, rack = Some((node % 2).toString))
```

Comment:
nit: since we're modifying this method already, maybe we can clean up the definition:
```scala
  override def generateConfigs: Seq[KafkaConfig] = {
    ..
  }
```